aeon_core_v1_2.py

```
@@ -0,0 +1,294 @@
+
+    def query_intent(self, state: np.ndarray, intent: np.ndarray, beta: float = 0.5,
+                    threshold: float = 0.7, nu: float = 0.05, delta: float = 0.1,
+                    epsilon: float = 0.05) -> tuple[np.ndarray, List[str]]:
+        """Query graph for intent update."""
+        sims = sorted([(n, np.dot(state, d["vector"]) / (np.linalg.norm(state) * np.linalg.norm(d["vector"]) + 1e-12))
+                       for n, d in self.G.nodes(data=True) if d["type"] == "State"],
+                      key=lambda x: x[1], reverse=True)[:5]
+
+        candidates = []
+        trace_log = []
+        for sim_node, sim in sims:
```

Ruff (0.12.2)

Loop control variable `sim` not used within loop body

Rename unused `sim` to `_sim`

(B007)